README.md

```diff
@@ -1,3 +1,39 @@
 # Ophan GeoIP DB Refresher
 
+_Retrieves regular updates of the [MaxMind GeoIP Database](https://www.maxmind.com/en/geoip2-databases)
+and uploads them to S3 for Ophan_
 
+Ophan uses geolocation by IP address to work out where in the world our readers are.
+[MaxMind](https://www.maxmind.com/) provide the proprietary GeoIP database we use to
+perform that lookup.
+
+## Steps performed by the Refresher
+
+* Download the database archive file (`.tar.gz`) from the MaxMind permalink url, along
+  with the archive's SHA256 hash (`.tar.gz.sha256`).
+* Compute the SHA256 hash of the downloaded archive, verify it matches the stated hash.
+* Inspect the database archive file to find archive entry that corresponds to the
+  database file (`GeoIP2-City.mmdb`) itself.
+* Stream the archive entry directly from the archive up to S3.
+
+## Running the Refresher locally
+
+### Pre-requisites
+
+These mostly match [the pre-requisites for running Ophan locally](https://github.com/guardian/ophan/blob/main/docs/developing-ophan/running-ophan-locally.md#pre-requisites) -
+specifically Java 11 & `sbt`, but also especially the requirement to have `ophan` AWS credentials
+from [Janus](https://janus.gutools.co.uk/).
+
+### Running the Lambda locally
+
+Execute this on the command line:
+
+```bash
+$ sbt run
+```
```

Comment:
Is this accurate:
- puts `test-GeoIP2-City.mmdb` into s3://ophan-dist/geoip
- `GeoIP2-City.mmdb` is what's used in PROD
- `GeoIP2-City.mmdb` is only updated on a schedule
- So then if running this locally fails, it doesn't affect geoip data in PROD

?

Reply:
Actually, running locally does exactly what the PROD lambda does, and it's now using `GeoIP2-City.mmdb` without any `test-` prefix. So, in principle, if the file was corrupted when running locally, then that could affect geoip data in PROD, which is not _ideal_. It would be reasonable I think to add some code so that when invoked by the CLI, it goes off to a test path.

However, the downloaded archive _is_ verified against hash, and _"Amazon S3 [never adds partial objects](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html); if you receive a success response, Amazon S3 added the entire object to the bucket."_ - so it's hard to see how it could go wrong in such a way as to corrupt the uploaded file, unless the code itself was unintentionally modified in some way to corrupt the file!

Reply:
Huh! I must have forgotten to pull before I tried running it because I ended up updating `test-GeoIP2-City.mmdb`. Good to know, thanks.

Reply:
No worries! I've just deleted that `test-GeoIP2-City.mmdb` file from S3 to avoid confusion going forward, the only file left is the official one:

![image](https://user-images.githubusercontent.com/52038/119492680-42799200-bd57-11eb-96de-f2fd535e6900.png)